src/skydb.ts

```diff
@@ -270,40 +257,55 @@ export async function setDataLink(
 
   const { publicKey: publicKeyArray } = sign.keyPair.fromSecretKey(hexToUint8Array(privateKey));
 
-  const entry = await getDataLinkRegistryEntry(this, toHexString(publicKeyArray), dataKey, dataLink, opts);
+  const getEntryOpts = extractOptions(opts, defaultGetEntryOptions);
+  const entry = await getNextRegistryEntry(
+    this,
+    toHexString(publicKeyArray),
+    dataKey,
+    decodeSkylink(dataLink),
+    getEntryOpts
+  );
 
   // Update the registry.
   const setEntryOpts = extractOptions(opts, defaultSetEntryOptions);
   await this.registry.setEntry(privateKey, entry, setEntryOpts);
 }
 
-export async function getDataLinkRegistryEntry(
+/**
+ * Gets the next entry for the given public key and data key, setting the data to be the given data and the revision number accordingly.
+ *
+ * @param client - The Skynet client.
+ * @param publicKey - The user public key.
+ * @param dataKey - The dat akey.
+ * @param data - The data to set.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @returns - The registry entry and corresponding data link.
+ * @throws - Will throw if the revision is already the maximum value.
+ */
+export async function getNextRegistryEntry(
```

Comment:
Something about this method's name bothers me. We're "building" the next registry entry? Or preparing it? But this feels like we're fetching it, or iterating through a collection.

Reply:
(This method might not even be needed after that refactor.)